cassandra/pool.py

```diff
@@ -360,18 +363,45 @@ def __init__(self, host, host_distance, session):
             return
 
         log.debug("Initializing connection for host %s", self.host)
-        self._connection = session.cluster.connection_factory(host.endpoint)
+        first_connection = session.cluster.connection_factory(host.endpoint)
+        log.debug("first connection created for shard_id=%i", first_connection.shard_id)
+        self._connections[first_connection.shard_id] = first_connection
         self._keyspace = session.keyspace
+
         if self._keyspace:
-            self._connection.set_keyspace_blocking(self._keyspace)
+            first_connection.set_keyspace_blocking(self._keyspace)
+
+        if first_connection.sharding_info:
+            self.host.sharding_info = weakref.proxy(first_connection.sharding_info)
+            for _ in range(first_connection.sharding_info.shards_count * 2):
+                conn = self._session.cluster.connection_factory(self.host.endpoint)
+                if conn.shard_id not in self._connections.keys():
+                    log.debug("new connection created for shard_id=%i", conn.shard_id)
+                    self._connections[conn.shard_id] = conn
+                    if self._keyspace:
+                        self._connections[conn.shard_id].set_keyspace_blocking(self._keyspace)
+
+                if len(self._connections.keys()) == first_connection.sharding_info.shards_count:
+                    break
+            if not len(self._connections.keys()) == first_connection.sharding_info.shards_count:
+                raise NoConnectionsAvailable("not enough shard connection opened")
```

Comment:
```suggestion
                missing_shards = first_connection.sharding_info.shards_count - len(self._connections.keys())
                raise NoConnectionsAvailable("Missing open connections to %i shards on host %s" % (missing_shards, self.host))
```

Reply:
excellent point, for whom would need to debug failures with it. would be priceless info,

will fix it, for future us :)